### ISSUE TITLE
Bug: Cannot find export error for glob exports

### ISSUE THREAD

**Issue by luismeyer** [FormidableLabs/trace-deps#70]
Hi, i'm using this lib as a part of serverless-jetpack and just briefly looked in the code so i'm not sure if this is the correct place to raise my concern. 

The problem i'm running into: 
`Error: Cannot find export './middleware.js.mjs' in module './middleware.js' from './node_modules/graphql-middleware/dist'.`

The library responsible for this error is graphql-middleware. The exports in the package.json looks like this:
```json
"exports": {
  "./*": {
    "require": "./*.js",
    "import": "./*.mjs"
  },
}
```

The files located in the dist dir of graphql-middleware are actually just called `middleware.mjs`. 
As far as i can see the resolve.exports lib returns the wrong relative paths. But im not sure if some configuration mistake on my side is the problem or if this isn't working properly.

**Comment by ryan-roemer** [comment]
Thanks. I'll take a look and see if I can reproduce as a regression test. `"./*": "./*.js"` is already a special case in real life Node runtime because it somewhat disables modern loading behavior and is not recommend. Probably the same for `./*.mjs`

**Comment by yankovalera** [comment]
Hi, I've managed to reproduce this. `resolve.exports` is resolving validation.js as ./validation.js.mjs -- see added tests [here](https://github.com/lukeed/resolve.exports/blob/0b69d12a22d68a7588129b154f39490cd5514088/test/resolve.js#L273)

**Comment by ryan-roemer** [comment]
Bug confirmed. I have a failing regression test at: https://github.com/FormidableLabs/trace-deps/compare/bug/wildcard-mjs

The _internal_ dependencies (in my tests `import { sub } from "./sub.mjs"`) should not be trying to further suffix from the `package.json:exports` wildcard. Failing test current status:

```sh
$ yarn test --grep suffixed
yarn run v1.22.11
$ mocha --grep suffixed


  lib/trace
    traceFile
      modern ESM exports
        complicated exports
          suffixed .mjs/.js/.cjs local paths with wildcards
            1) handles CJS static imports
            2) handles ESM static imports


  0 passing (26ms)
  2 failing

  1) lib/trace
       traceFile
         modern ESM exports
           complicated exports
             suffixed .mjs/.js/.cjs local paths with wildcards
               handles CJS static imports:
     Error: Encountered resolution error in /Users/rye/scm/fmd/trace-deps/node_modules/pkg/index.cjs for ./sub.cjs: Error: Cannot find export './sub.cjs.mjs' in module './sub.cjs' from '/Users/rye/scm/fmd/trace-deps/node_modules/pkg'
      at /Users/rye/scm/fmd/trace-deps/lib/trace.js:425:13
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at async Promise.all (index 0)
      at async traceFile (lib/trace.js:540:19)
      at async _recurseDeps (lib/trace.js:181:22)
      at async traceFile (lib/trace.js:571:20)
      at async Context.<anonymous> (test/lib/trace.spec.js:1921:48)

  2) lib/trace
       traceFile
         modern ESM exports
           complicated exports
             suffixed .mjs/.js/.cjs local paths with wildcards
               handles ESM static imports:
     Error: Encountered resolution error in /Users/rye/scm/fmd/trace-deps/node_modules/pkg/index.cjs for ./sub.cjs: Error: Cannot find export './sub.cjs.mjs' in module './sub.cjs' from '/Users/rye/scm/fmd/trace-deps/node_modules/pkg'
      at /Users/rye/scm/fmd/trace-deps/lib/trace.js:425:13
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at async Promise.all (index 0)
      at async traceFile (lib/trace.js:540:19)
      at async _recurseDeps (lib/trace.js:181:22)
      at async traceFile (lib/trace.js:571:20)
      at async Context.<anonymous> (test/lib/trace.spec.js:1921:48)
```

(Test still needs additional files in expect to eventually pass, but should not bail with error first as seen here).

**Comment by ryan-roemer** [comment]
Update -- I have a WIP branch to test out @luismeyer if you want to kick the tires!

https://github.com/FormidableLabs/trace-deps/compare/bug/wildcard-mjs

**Comment by luismeyer** [comment]
Thanks. It works for me

**Comment by ryan-roemer** [comment]
Fix released in `trace-deps@0.4.4`.
